src/Stratis.SmartContracts.Executor.Reflection/CallData.cs

```diff
@@ -1,74 +1,23 @@
 ﻿using NBitcoin;
-using Stratis.SmartContracts.Core;
 
 namespace Stratis.SmartContracts.Executor.Reflection
 {
-    public struct CallData
+    /// <summary>
+    /// Smart contract transaction data for a method call
+    /// </summary>
+    public class CallData : ICallData
     {
-        public CallData(int vmVersion, ulong gasPrice, Gas gasLimit, uint160 address,
-            string method, string rawParameters = "", object[] methodParameters = null)
+        public CallData(Gas gasLimit, uint160 address, string method, object[] methodParameters = null)
```

Comment:
This might be trivial but can we name address here `contractAddress`? I always confused with what address we refer to as and when.

Reply:
Also, please move the properties to the top of the constructor so that its inline with FN.